### ISSUE TITLE
Optimize Media Startup

### ISSUE THREAD

**Issue by mraavi** [google/shaka-player#2306]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
YES

**What version of Shaka Player are you using?**
2.5.3

**Can you reproduce the issue with our latest release version?**
YES

**Can you reproduce the issue with the latest code from `master`?**
YES

**What browser and OS are you using?**
Chrome / MAC

Our application stream Live media.
Media Start up time is 4 to 5 sec.
I am looking for ways to reduce media start time up time in order to provide best user experience.
Following are the list of call during media start up.

<img width="333" alt="image" src="https://user-images.githubusercontent.com/6366393/70593454-318c9d80-1c03-11ea-8f51-4ff9a1d7a99f.png">

You can observe that the call in the list
MPD
init.mp4 for audio
1st audio segment
init.mp4 for video
1st video segment
get License call
2nd audio segment
3rd audio segment
2nd video segment
Media start only after downloading 2nd video segment. Segment duration is 6 sec.
Following are the abr and streaming configurations
```js
mPlayer.configure({
            abr: {
                defaultBandwidthEstimate: 1000000,
                switchInterval: 6
            },
            streaming: {
                bufferBehind: 6,
                bufferingGoal: 6
            }
        });
```

Following is the mpd,

```
<?xml version="1.0" encoding="utf-8"?>
<MPD xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:mobitv="urn:mobitv" xmlns="urn:mpeg:dash:schema:mpd:2011" xsi:schemaLocation="urn:mpeg:dash:schema:mpd:2011 DASH-MPD.xsd" profiles="urn:mpeg:dash:profile:isoff-live:2011,urn:com:dashif:dash264" maxSegmentDuration="PT6S" minBufferTime="PT2S" type="dynamic" timeShiftBufferDepth="PT20M" minimumUpdatePeriod="PT8760H" availabilityStartTime="1970-01-01T00:00:19Z" suggestedPresentationDelay="PT11S" id="5062" publishTime="2019-12-11T05:23:54Z">
   <ProgramInformation>
      <Title>Media Presentation Description by MobiTV.</Title>
   </ProgramInformation>
   <Period mobitv:publishTime="2019-12-11T05:23:54Z" id="precambrian" start="PT0S" mobitv:thumbnails="150_T;640_T">
      <AdaptationSet contentType="video" mimeType="video/mp4" mobitv:qualityLevel="SD" segmentAlignment="true" startWithSAP="1" par="16:9" minWidth="640" maxWidth="1280" minHeight="360" maxHeight="720" maxFrameRate="60/2">
         <Accessibility schemeIdUri="urn:scte:dash:cc:cea-608:2015"/>
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="main"/>
         <SegmentTemplate startNumber="0" initialization="$RepresentationID$/init.mp4" duration="6" media="$RepresentationID$/$Number$.m4s"/>
         <ContentProtection schemeIdUri="urn:mpeg:dash:mp4protection:2011" value="cenc"/>
         <ContentProtection schemeIdUri="urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed" value="Widevine: SD"/>
         <ContentProtection schemeIdUri="urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95" value="Playready: SD"/>
         <Representation id="V2200_W" codecs="avc1.4d401f" bandwidth="2384800" mobitv:average-bandwidth="2205862" mobitv:peak-bandwidth="2361517" width="960" height="540" frameRate="60/2" sar="1:1"/>
         <Representation id="V5000_W" codecs="avc1.640020" bandwidth="5120000" mobitv:average-bandwidth="5001378" mobitv:peak-bandwidth="5284876" width="1280" height="720" frameRate="60/2" sar="1:1"/>
         <Representation id="V864_W" codecs="avc1.4d401e" bandwidth="884736" mobitv:average-bandwidth="871321" mobitv:peak-bandwidth="944708" width="640" height="360" frameRate="60/2" sar="1:1"/>
      </AdaptationSet>
      <AdaptationSet contentType="audio" mimeType="audio/mp4" lang="eng" mobitv:qualityLevel="SD" segmentAlignment="true" startWithSAP="1">
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="main"/>
         <SegmentTemplate startNumber="0" initialization="$RepresentationID$/init.mp4" duration="6" media="$RepresentationID$/$Number$.m4s"/>
         <ContentProtection schemeIdUri="urn:mpeg:dash:mp4protection:2011" value="cenc"/>
         <ContentProtection schemeIdUri="urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed" value="Widevine: SD"/>
         <ContentProtection schemeIdUri="urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95" value="Playready: SD"/>
         <Representation id="A96_ENG" codecs="mp4a.40.5" bandwidth="98304" mobitv:average-bandwidth="101359" mobitv:peak-bandwidth="102064" audioSamplingRate="48000">
            <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
         </Representation>
         <InbandEventStream schemeIdUri="urn:mobitv:event" value="id3"/>
      </AdaptationSet>
   </Period>
</MPD>
```

How to start the media right after LICENSE response? First audio and video segments are downloaded before the License call . why to wait till 2nd video segment download ?
i tried setting bufferingGoal lower than 6 sec. That didn't help.

**Comment by theodab** [comment]
Please don't open multiple github issues for the same issue. This is a duplicate of #2291.
